### PR TITLE
Fix issues with RLP enconding transactions, receipts, and headers

### PR DESCRIPTION
**Pull request by severiano-sisneros**
The primary change was converting substream types into reth-primitives types (transactions, receipts, and block header). 

RLP encoding was then handled by reth code.

### CODE REVIEW

File: rust-toolchain.toml
Comment (taturosati) [inline rust-toolchain.toml:1]: is nightly channel necessary?

File: src/e2store/mod.rs [hunk 1]
```diff
@@ -220,6 +255,24 @@ impl TryFrom<BlockHeader> for E2Store {
...
+        Ok(E2Store {
+            type_: E2StoreType::CompressedHeader,
+            length: data.len() as u32,
+            reserved: 0,
+            data,
+        })
+    }
+}
+
 impl TryFrom<BlockBody> for E2Store {
```
Comment (taturosati) [inline src/e2store/mod.rs]: Is this code used? i would consider removing it

File: src/e2store/mod.rs [hunk 2]
```diff
@@ -266,3 +380,323 @@ pub fn encode_bigint(big_int: BigInt) -> Vec<u8> {
 
     bytes
 }
+
+fn map_success(status: &i32) -> Result<bool, ReceiptError> {
```
Comment (taturosati) [inline src/e2store/mod.rs]: Why return result if it can never fail?

File: src/e2store/rlp_impl.rs [hunk 3]
```diff
@@ -54,50 +45,54 @@ impl Encodable for BlockBody {
 
 impl Encodable for Transaction {
```
Comment (taturosati) [inline src/e2store/rlp_impl.rs]: I dont think this code is used now that we are using reth encoding

File: rust-toolchain.toml
Comment (severiano-sisneros) [reply rust-toolchain.toml:1]: It was necessary at one point, but it looks like it's not necessary anymore. We can set it to a stable.

File: src/e2store/mod.rs (hunk 1 above)
Comment (severiano-sisneros) [reply src/e2store/mod.rs]: Removed in [3a613b0](https://github.com/semiotic-ai/era-file-sink/pull/1/commits/3a613b039e521607ea5e3ca4f07acd4b487efcfa)

File: src/e2store/rlp_impl.rs (hunk 3 above)
Comment (severiano-sisneros) [reply src/e2store/rlp_impl.rs]: Removed in [3a613b0](https://github.com/semiotic-ai/era-file-sink/pull/1/commits/3a613b039e521607ea5e3ca4f07acd4b487efcfa)

File: src/e2store/mod.rs (hunk 2 above)
Comment (severiano-sisneros) [reply src/e2store/mod.rs]: Retuning bool in [f438e40](https://github.com/semiotic-ai/era-file-sink/pull/1/commits/f438e40e124a3b80d2f9c65af9de91fc7e550639)

File: rust-toolchain.toml
Comment (severiano-sisneros) [reply rust-toolchain.toml:1]: I just removed the channel requirement in [573cf43](https://github.com/semiotic-ai/era-file-sink/pull/1/commits/573cf433339ef648f714df81128fb76c5ab94cbe)

**Pull request merged**